thirdeye-server/src/main/java/ai/startree/thirdeye/resources/RcaInvestigationResource.java

```diff
@@ -48,21 +54,33 @@ public class RcaInvestigationResource extends CrudResource<RcaInvestigationApi,
 
   @Inject
   public RcaInvestigationResource(final RcaInvestigationManager rootCauseSessionDAO,
+      final AnomalyManager anomalyManager,
       final AuthorizationManager authorizationManager) {
     super(rootCauseSessionDAO, API_TO_INDEX_FILTER_MAP, authorizationManager);
+    this.anomalyManager = anomalyManager;
   }
 
   @Override
   protected RcaInvestigationDTO createDto(final ThirdEyePrincipal principal,
       final RcaInvestigationApi api) {
-    final RcaInvestigationDTO rcaInvestigationDTO = ApiBeanMapper.toDto(api);
+    final RcaInvestigationDTO rcaInvestigationDTO = toDto(api);
     rcaInvestigationDTO.setCreatedBy(principal.getName());
     return rcaInvestigationDTO;
   }
 
   @Override
   protected RcaInvestigationDTO toDto(final RcaInvestigationApi api) {
-    return ApiBeanMapper.toDto(api);
+    final RcaInvestigationDTO dto = ApiBeanMapper.toDto(api);
+
+    // Copy auth from the anomaly.
+    if (dto.getAuth() == null) {
```

Comment:
this is tricky stuff. This is required for all entities. no?

Reply:
I'm not sure I understand. What other entities do you have in mind?

Reply:
A safer way to do this might be to have the UI copy the namespace from the anomaly and send it in the POST request, but I'm not sure how quickly that can be done.

Reply:
So, this is used for create/edit investigations right? so, can I set the auth to something and pass in? then that is respected instead of what is set in anomaly?

Reply:
Ya. if there's no auth config from the api object, then it receives a copy from the anomaly.